packages/react-components/src/gallery/dominantSpeaker.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { VideoGalleryRemoteParticipant } from '../types';
+
+/**
+ * Calculates the participants that should be renderd basis on the list of dominant
+ * speakers and currently renderd participants in a call.
+ * @param participants - Array containing all participants of a call.
+ * {@link @azure/communication-react#VideoGalleryRemoteParticipant}
+ * @param dominantSpeakers - An array containing the userId of dominant speakers
+ * in a call in the order of their dominance. 0th index is the most dominant, 1st is the second most etc
+ * @param visibleParticipants - Array containing currently rendered (visible)
+ * participants in the call. {@link @azure/communication-react#VideoGalleryRemoteParticipant}
+ * @param maxTiles - Maximum number of tiles to calculate.
+ * @returns VideoGalleryRemoteParticipant[] {@link @azure/communication-react#VideoGalleryRemoteParticipant}
+ */
+export const smartDominantSpeakerParticipants = (
+  participants: VideoGalleryRemoteParticipant[] = [],
+  dominantSpeakers: Array<string> = [],
+  visibleParticipants: VideoGalleryRemoteParticipant[] = [],
+  maxTiles = 4
```

Comment:
This number is a restriction from SDK or just our design?

Reply:
nit: It's bit of a faff but could it alos be hoisted to the top of the file as `const MAX_TILES_SUPPORTED = 4` with the comment about the SDk limitation - just makes it super clear and easy to change (person changing it knows they only have to change one place and haven't missed somewhere)

Reply:
This function needs to have it as a parameter since this limit applies to `video` tiles and not to `audio` tiles.
We can render 100s of participants that have their video turned off and a component will need to specify that to this function.

Reply:
Oh yep keep it as a param, just have `maxTiles = MAX_TILES_SUPPORTED`

Reply:
Haha, I had it earlier as `MAX_TILES_SUPPORTED` but @prprabhu-ms  advised I change it to `maxTiles`
It's a matter of standard I guess.
I am ok either ways...

But given that it's a function argument, `maxTiles` makes more sense...